mmseg/models/backbones/fast_scnn.py

```diff
@@ -0,0 +1,361 @@
+from mmseg.ops import DepthwiseSeparableConvModule, resize
+from ..builder import BACKBONES
+
+
+class LearningToDownsample(nn.Module):
+    """Learning to downsample module.
+    Args:
+        in_channels (int): Number of input channels.
+
+        dw_channels1 (int): Number of output channels of the first
+            depthwise conv (dwconv) layer.
+
+        dw_channels2 (int): Number of output channels of the second
+            dwconv layer.
+
+        out_channels (int): Number of output channels of the whole
+            'learning to downsample' module.
+
+
+        conv_cfg (dict|None): Config of conv layers.
+        norm_cfg (dict|None): Config of norm layers.
+        act_cfg (dict): Config of activation layers.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 dw_channels1,
+                 dw_channels2,
+                 out_channels,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN'),
+                 act_cfg=dict(type='ReLU')):
+        super(LearningToDownsample, self).__init__()
+        self.conv_cfg = conv_cfg
+        self.norm_cfg = norm_cfg
+        self.act_cfg = act_cfg
+        self.conv = ConvModule(
+            in_channels,
+            dw_channels1,
+            3,
+            stride=2,
+            conv_cfg=self.conv_cfg,
+            norm_cfg=self.norm_cfg,
+            act_cfg=self.act_cfg)
+        self.dsconv1 = DepthwiseSeparableConvModule(
+            dw_channels1,
+            dw_channels2,
+            kernel_size=3,
+            stride=2,
+            padding=1,
+            norm_cfg=self.norm_cfg)
+        self.dsconv2 = DepthwiseSeparableConvModule(
+            dw_channels2,
+            out_channels,
+            kernel_size=3,
+            stride=2,
+            padding=1,
+            norm_cfg=self.norm_cfg)
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.dsconv1(x)
+        x = self.dsconv2(x)
+        return x
+
+
+class GlobalFeatureExtractor(nn.Module):
+    """Global feature extractor module.
+    Args:
+        in_channels (int): Number of input channels of the GFE module.
+
+        block_channels (tuple): Tuple of ints. Each int specifies the
+            number of output channels of each Inverted Residual module.
+
+        out_channels(int): Number of output channels of the GFE module.
+
+        t (int): t parameter (upsampling factor) of each Inverted Residual
+            module.
+
+        num_blocks (tuple): Tuple of ints. Each int specifies the number of
+        times each Inverted Residual module is repeated.
+
+        pool_scales (tuple): Tuple of ints. Each int specifies the parameter
+            required in 'global average pooling' within PPM.
+
+        conv_cfg (dict|None): Config of conv layers.
+        norm_cfg (dict|None): Config of norm layers.
+        act_cfg (dict): Config of activation layers.
+        align_corners (bool): align_corners argument of F.interpolate.
```

Comment:
No newline is needed between arguments. 
Default value is missing.

Reply:
Fixed.